x-pack/plugins/data_visualizer/public/application/index_data_visualizer/hooks/esql/use_esql_overall_stats_data.ts

```diff
@@ -0,0 +1,395 @@
+  if (!isESQLQuery(query)) {
+    throw Error(
+      i18n.translate('xpack.dataVisualizer.esql.noQueryProvided', {
+        defaultMessage: 'No ES|QL query provided',
+      })
+    );
+  }
+  const esqlBaseQuery = query.esql;
+  let earliestMs = Infinity;
+  let latestMs = -Infinity;
+
+  if (timeFieldName) {
+    const aggQuery = ` | EVAL _timestamp_= TO_DOUBLE(DATE_TRUNC(${intervalMs} millisecond, ${getSafeESQLName(
+      timeFieldName
+    )}))
+    | stats rows = count(*) by _timestamp_
+    | LIMIT 10000`;
+
+    const request = {
+      params: {
+        query: esqlBaseQuery + aggQuery,
+        ...(filter ? { filter } : {}),
+      },
+    };
+    try {
+      const esqlResults = await runRequest(request, { ...(searchOptions ?? {}), strategy: 'esql' });
+      let totalCount = 0;
+      const _buckets: Record<string, number> = {};
+      // @ts-expect-error ES types needs to be updated with columns and values as part of esql response
+      esqlResults?.rawResponse.values.forEach((val) => {
+        const [count, bucket] = val;
+        _buckets[bucket] = count;
+        totalCount += count;
+        if (bucket < earliestMs) {
+          earliestMs = bucket;
+        }
+        if (bucket >= latestMs) {
+          latestMs = bucket;
+        }
+      });
+      const result: DocumentCountStats = {
+        interval: intervalMs,
+        probability: 1,
+        randomlySampled: false,
+        timeRangeEarliest: earliestMs,
+        timeRangeLatest: latestMs,
+        buckets: _buckets,
+        totalCount,
+      };
+      return { documentCountStats: result, totalCount };
+    } catch (error) {
+      handleError({
+        request,
+        error,
+        onError,
+        title: i18n.translate('xpack.dataVisualizer.esql.docCountError', {
+          defaultMessage: `Error getting total count & doc count chart for ES|QL time-series data for request:`,
+        }),
+      });
+      return Promise.reject(error);
+    }
+  } else {
+    //  If not time field, get the total count
+    const request = {
+      params: {
+        query: esqlBaseQuery + ' | STATS _count_ = COUNT(*)  | LIMIT 1',
+        ...(filter ? { filter } : {}),
+      },
+    };
+    try {
+      const esqlResults = await runRequest(request, { ...(searchOptions ?? {}), strategy: 'esql' });
+      return {
+        documentCountStats: undefined,
+        totalCount: esqlResults?.rawResponse.values[0][0],
+      };
+    } catch (error) {
+      handleError({
+        request,
+        error,
+        onError,
+        title: i18n.translate('xpack.dataVisualizer.esql.docCountNoneTimeseriesError', {
+          defaultMessage: `Error getting total count for ES|QL data:`,
+        }),
+      });
+      return Promise.reject(error);
+    }
+  }
+};
+
+export const getInitialData = (): Data => ({
+  timeFieldName: undefined,
+  columns: undefined,
+  totalCount: undefined,
+});
+
+const NON_AGGREGATABLE_FIELD_TYPES = new Set<string>([
```

Comment:
Do you know why the keyword `k` icon is showing up for these geo fields?

<img width="790" alt="image" src="https://github.com/elastic/kibana/assets/7405507/38e9e44d-52e3-40e3-b168-d1aa1c069229">